ansible_collections/arista/avd/docs/containers/overview.md

```diff
@@ -0,0 +1,92 @@
+
+## Why Dev Containers
+
+Building a functional Ansible environment can be a challenging task. A manually created Ansible environment can be broken due to number of reasons:
+
+- Problems with Python and Ansible versions
+- Broken dependencies
+- Interpreter path issues
+- Inconsistent Ansible configuration files
+- etc.
+
+Containers can provide a simple and portable way to build a functional, portable and isolated Ansible environment. [Dev Containers](https://containers.dev) bring that to the next level by simplifying container build process and extending the functionality. The [Dev Container Specification](https://github.com/devcontainers/spec) was started by Microsoft and has strong community support. Essentially Dev Containers are powered by following:
+
+- [Prebuilt images](https://github.com/devcontainers/images)
+- [Features](https://containers.dev/features)
+
+Dev containers can be run locally or remotely, in a private or public cloud, in a variety of [supporting tools and editors](https://containers.dev/supporting).
+
+## AVD Dev Container Variants
+
+The AVD repository provides a number of pre-build dev container images for various use cases:
+
+- **base** - image that is used to build other variants. It is not intended for direct use.
+- **dev** - an image created for AVD contributors and to support quick testing of AVD branches. The AVD collection must be mounted in this container or installed from a git branch.
+- **universal** - an image with pre-installed AVD collection and its dependencies that can be used to build and deploy AVD configuration using a specific AVD version.
+
+All images are based on [Python Slim](https://hub.docker.com/_/python) with some additional features and extensions pre-installed.
+
+## How to Use Dev Containers
+
+Currently the only supported option is using [VSCode](https://code.visualstudio.com/) to start a dev container.
+
+To start using a dev container:
+
+- Install [VScode Dev Containers extension](https://code.visualstudio.com/docs/devcontainers/tutorial)
+- Add `.devcontainer/devcontainer.json` to your AVD repository (check examples below for possible dev container definitions).
+- Open VSCode command pallette and pick `Dev Containers: Rebuild and Reopen in Container`
+- Wait until the build will finish and enjoy running AVD in a container.
+
+If you want to use a specific AVD release, use the following `devcontainer.json`:
+
+```json
+{
+    "name": "AVD Universal",
+    "image": "ghcr.io/aristanetworks/ansible-avd/universal:python3.11-avd-v4.4.0"
+}
+```
+
+For AVD contributors the following configuration is recommended:
+
+```json
+{
+    "name": "AVD Development",
+    "image": "ghcr.io/aristanetworks/ansible-avd/dev:python3.11",
+    "mounts": [
+        "source=${localWorkspaceFolder}/ansible_collections/arista/avd,target=/home/avd/.ansible/collections/ansible_collections/arista/avd,type=bind"
```

Comment:
So this is confusing to me. A developer would use the development container as part of the development workflow itself. But here you are using it for *running*  AVD with the local checked out git branch. That can work fine, but it will spin up two seperate container instances right?